src/mail_client.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABC, abstractmethod
+import boto3.session
+import boto3
+
+
+class MailServiceError(Exception):
+    def __init__(self, error, exp):

```

Comment:
Let's rename `error` to `message` and `exp` to `cause` (try SHIFT + F6 for renaming), it's easier to understand which is which.